zipline/assets/_assets.pyx

```diff
@@ -187,37 +201,58 @@ cdef class Asset:
         """
         return cls(**dict_)
 
-    def _is_alive(self, dt, bool normalized):
+    def _is_alive_for_session(self, session_label):
         """
         Returns whether the asset is alive at the given dt.
 
         Parameters
         ----------
-        dt: pd.Timestamp
-            The desired timestamp.
-
-        normalized: boolean
-            Whether the date has already been normalized.  If not, we need
-            to first normalize the date before doing the alive check.  If the
-            date is already normalized, this method runs up to 80% faster.
+        session_label: pd.Timestamp
+            The desired session label to check. (midnight UTC)
 
         Returns
         -------
         boolean: whether the asset is alive at the given dt.
         """
-        cdef int64_t dt_value
         cdef int64_t ref_start
         cdef int64_t ref_end
 
-        if not normalized:
-            dt_value = normalize_date(dt).value
-        else:
-            dt_value = dt.value
-
         ref_start = self.start_date.value
         ref_end = self.end_date.value
 
-        return ref_start <= dt_value <= ref_end
+        return ref_start <= session_label.value <= ref_end
+
+    def _asset_exchange_open(self, dt_minute):
+        """
+        Parameters
+        ----------
+        dt_minute: pd.Timestamp (UTC, tz-aware)
+            The minute to check.
+
+        Returns
+        -------
+        boolean: whether the asset's exchange is open at the given minute.
+        """
+        calendar = self._exchange_trading_calendar_for_asset()
+        return calendar.is_open_on_minute(dt_minute)
+
+    def _exchange_trading_calendar_for_asset(self):
+        """
+        Get the calendar for this asset's exchange.
+
+        Raises KeyError if the asset's exchange calendar cannot be found.
+
+        Returns
+        -------
+        The asset's exchange's trading calendar.
+        """
+        if isinstance(self, Equity):
+            # FIXME: probably too Quantopian-specific
+            return get_calendar("NYSE")
```

Comment:
yup.  The asset db now has the canonical exchange name ("NASDAQ", "CME", etc) and we use that to look up the calendar.